### PR TITLE
Add MinUtxoValue for ignoring spam

### LINKED ISSUE #186: Feature Request: Ignore small amounts (configurable)
Tracking and scanutxoset should remove UTXOs below a specified threshold.

Default is none.

Could be an environment/runtime parameter.

Use case: Do not want to accidentally spend those spammy 1 satoshi inputs or recognize utxos that are below minimum deposit limit.

This is a very common metric for keeping spam from clogging up our systems.

### PR DESCRIPTION
**Pull request by junderw**
Fixes #186 (Maybe??? lol)

Am I heading in the right direction here?

### CODE REVIEW

**Comment by junderw** [comment]
Will this also block spam unconfirmed transactions as well? (I noticed it was talking about blockId)

**Comment by junderw** [comment]
I should also add stuff to SCANUtxoSetService and LiquidRepository

**Comment by junderw** [comment]
I should modify `SaveMatches` instead of modifying SCANUtxo. SCANUtxo can report 200 matches found and if 50 of them are spam, there will only be 150 added to the repo.

**Comment by junderw** [comment]
actually... SCANUtxo is too much of a pain... I think this PR is fine as is.

**Comment by junderw** [comment]
`I think I actually need to modify `UpdateRepository` in SCANUtxoSetService... so I will give up.

File: NBXplorer/Repository.cs [hunk 1]
```diff
@@ -735,6 +736,10 @@ public int MaxPoolSize
 		{
 			get; set;
 		} = 30;
+		public int MinUtxoValue
```
Comment (NicolasDorier) [inline NBXplorer/Repository.cs]: Change `int` to `Money`.

File: NBXplorer/Configuration/ExplorerConfiguration.cs [hunk 2]
```diff
@@ -135,6 +140,7 @@ public ExplorerConfiguration LoadArgs(IConfiguration config)
 					}
 
 					chainConfiguration.StartHeight = config.GetOrDefault<int>($"{network.CryptoCode}.startheight", -1);
+					chainConfiguration.MinUtxoValue = config.GetOrDefault<int>($"{network.CryptoCode}.minutxovalue", 1);
```
Comment (NicolasDorier) [inline NBXplorer/Configuration/ExplorerConfiguration.cs]: You can convert `int` to `Money` with `Money.Satoshis(value)`

**Comment by junderw** [comment]
Please review the new commit. Thanks.

**Comment by junderw** [comment]
Question: Would the following also fix ScanUtxoSet???

```diff
diff --git a/NBXplorer/ScanUTXOSetService.cs b/NBXplorer/ScanUTXOSetService.cs
index bf87d26..52206ac 100644
--- a/NBXplorer/ScanUTXOSetService.cs
+++ b/NBXplorer/ScanUTXOSetService.cs
@@ -249,7 +249,7 @@ namespace NBXplorer
                                                          BlockId: blockIdsByHeight.TryGet(o.First().Height),
                                                          TxId: o.Select(c => c.Coin.Outpoint.Hash).FirstOrDefault(),
                                                          KeyPathInformations: o.Select(c => scannedItems.KeyPathInformations[c.Coin.ScriptPubKey]).ToList()))
-                               .Where(o => o.BlockId != null)
+                               .Where(o => o.BlockId != null && o.Coin.Amount.Satoshis >= repo.MinUtxoValue)
                                .Select(o =>
                                {
                                        foreach (var keyInfo in o.KeyPathInformations)
```

**Comment by junderw** [comment]
Also...

I noticed that this PR and the ScanUTXOSet diff I make up there seems to talk about blocks...

Does this merged PR also filter out unconfirmed UTXOs by value as well? Or would I need to fix something else?

**Pull request merged**